src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+using Microsoft.VisualStudio.TestPlatform.Client;
+using Microsoft.VisualStudio.TestPlatform.Client.DesignMode;
+using Microsoft.VisualStudio.TestPlatform.Client.RequestHelper;
+using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers;
+using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Tracing;
+using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Tracing.Interfaces;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Interfaces;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Payloads;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using Microsoft.VisualStudio.TestPlatform.VsTestConsole.TranslationLayer.Interfaces;
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine;
+
+/// <summary>
+/// The in-process wrapper.
+/// </summary>
+internal class InProcessVsTestConsoleWrapper : IVsTestConsoleWrapper
+{
+    // Must be in sync with the highest supported version in
+    // src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs file.
+    private readonly int _highestSupportedVersion = 6;
+
+    private readonly ITranslationLayerRequestSender _requestSender;
+    private readonly ITestPlatformEventSource _testPlatformEventSource;
+    private bool _isInitialized;
+    private bool _sessionStarted;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="InProcessVsTestConsoleWrapper"/>.
+    /// </summary>
+    /// 
+    /// <param name="consoleParameters">The console parameters.</param>
+    public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
+        : this(
+              consoleParameters,
+              requestSender: new VsTestConsoleRequestSender(),
+              testRequestManager: null,
+              executor: new Executor(ConsoleOutput.Instance),
+              testPlatformEventSource: TestPlatformEventSource.Instance)
+    { }
+
+    internal InProcessVsTestConsoleWrapper(
+        ConsoleParameters consoleParameters,
+        ITranslationLayerRequestSender requestSender,
+        ITestRequestManager? testRequestManager,
+        Executor executor,
+        ITestPlatformEventSource testPlatformEventSource)
+    {
+        EqtTrace.Info("VsTestConsoleWrapper.StartSession: Starting VsTestConsoleWrapper session.");
+
+        _testPlatformEventSource = testPlatformEventSource;
+        _testPlatformEventSource.TranslationLayerInitializeStart();
+
+        // Start communication.
+        _requestSender = requestSender;
+        var port = _requestSender.InitializeCommunication();
+        if (port <= 0)
+        {
+            // Close the sender as it failed to host server.
+            _requestSender.Close();
+            throw new TransationLayerException("Error hosting communication channel.");
```

Comment:
I guess we need a resx entry here.

Reply:
Fixed.

Reply:
Is it missing a push? I still see the string entry here.